src/http.cpp

```diff
@@ -977,7 +995,23 @@ Request::handle_response_body(const asio::error_code& ec, const size_t bytes)
             std::bind(&Request::handle_response_body, this, std::placeholders::_1, std::placeholders::_2));
     }
     else if (connection_type_ == restinio::http_connection_header_t::close)
+    {
         terminate(asio::error::eof);
+    }
+    else if (response_.headers[HTTP_HEADER_TRANSFER_ENCODING] == HTTP_HEADER_TRANSFER_ENCODING_CHUNKED)
+    {
+        std::istream is(&conn_->data());
+        std::string chunk_size;
+        std::getline(is, chunk_size);
+        if (chunk_size.size() == 0){
+            std::replace(response_.body.begin(), response_.body.end(), '\n', '\0');
```

Comment:
What does this do ?

Reply:
it ensures that if there is no content-length following a body read aka the line is empty, we will not continue reading and consider that we received all of the chunks. The server is normally suppose to send 0\r\n which our tomcat of jams did but over the weekend only the 8080 was available and if you perform a curl on the functionality of chunked encoding while getting the devices, it does not do that and simply finishes by not writing nothing.